src/Libraries/RevitNodes/GeometryConversion/ProtoToRevitFace.cs

```diff
@@ -0,0 +1,154 @@
+
+               var ngeom = geom.ToNurbsSurface();
+               bool flipped = false;
+
+               // Check if the nurbs surface has flipped compared to the original surface
+               if (geom.NormalAtParameter(.5, .5).Dot(ngeom.NormalAtParameter(.5, .5)) < 0)
+               {
+                   flipped = true;
+               }
+
+               // Create Revit nurbs surface
+               var bbface = BRepBuilderSurfaceGeometry.CreateNURBSSurface(ngeom.DegreeU, ngeom.DegreeV,
+                   ngeom.UKnots(), ngeom.VKnots(), ngeom.ControlPoints().SelectMany(x => x.Select(y => y.ToXyz())).ToList(),
+                   ngeom.Weights().SelectMany(x => x).ToList(),
+                   false,
+                   new BoundingBoxUV());
+
+               // Dispose geometries
+               geom.Dispose();
+               ngeom.Dispose();
+
+               // Add face
+               var faceId = brb.AddFace(bbface, flipped);
+
+               // add loops and connected edges
+               foreach (var loop in protoFace.Loops)
+               {
+                   var loopId = brb.AddLoop(faceId);
+
+                   foreach (var coedge in loop.CoEdges)
+                   {
+                       var edge = coedge.Edge;
+                       BRepBuilderGeometryId edgeId;
+                       if (edgeDict.ContainsKey(edge))
+                       {
+                           edgeId = edgeDict[edge];
+                       }
+                       else
+                       {
+                           var curve = edge.CurveGeometry;
+                           // Revit is already projecting edges onto the surface after checking for loop consistency
+                           // and is quite forgiving even when we use the nurbs surface instead of 
+                           // the original surface.
+                           //
+                           // But there are cases when edges ends up slightly outside one of the surfaces and Revit fails.
+                           //
+                           // This is something that we can be improve going forward.
+                           edgeId = brb.AddEdge(BRepBuilderEdgeGeometry.Create(curve.ToRevitType()));
+                           edgeDict[edge] = edgeId;
+                       }
+                       brb.AddCoEdge(loopId, edgeId, coedge.Reversed);
+                   }
+
+                   brb.FinishLoop(loopId);
+               }
+
+               brb.FinishFace(faceId);
+           }
+
+           //clean up everything
+           edgeDict.ToList().ForEach(x => x.Key.Dispose());
+           faces.ForEach(x => x.Dispose());
+
+           // Get result
+           var outcome = brb.Finish();
+           var converted = brb.GetResult();
+
+           if (converted == null)
+           {
+               throw new Exception("An unexpected failure occurred when attempting to convert the solid");
```

Comment:
I guess it could for a node developer as it is right now. I will add resources for it.